coordinator/tests/services/translation/line-item.spec.ts

```diff
@@ -131,3 +134,64 @@ describe("additionalInstructions", () => {
     expect(result.pertinentInformation1.pertinentAdditionalInstructions.value).toBe(`${patientInfo}CD: ${exampleControlledDrugString}\n${patientInstruction}`)
   })
 })
+
+describe("prescriptionEndorsements", () => {
+  let bundle: fhir.Bundle
+  let firstFhirMedicationRequest: fhir.MedicationRequest
+
+  beforeEach(() => {
+    bundle = clone(TestResources.examplePrescription1.fhirMessageUnsigned)
+    firstFhirMedicationRequest = getMedicationRequests(bundle)[0]
+  })
+
+  beforeEach(() => {
+    bundle = clone(TestResources.examplePrescription1.fhirMessageUnsigned)
+  })
+
+  test("are translated when present", () => {
+    const medicationRequests = getMedicationRequests(bundle)
+
+    const prescriptionEndorsements = medicationRequests
+      .map(medicationRequest =>
+        getExtensionForUrlOrNull(medicationRequest.extension, "https://fhir.nhs.uk/R4/StructureDefinition/Extension-PrescriptionEndorsement") as fhir.CodeableConceptExtension)
+
+    expect(prescriptionEndorsements.length).toBeGreaterThan(0)
+
+    prescriptionEndorsements.map(prescriptionEndorsement =>
+      expect(prescriptionEndorsement.valueCodeableConcept.coding.length).toBeGreaterThan(0)
+    )
+
+    const hl7v3LineItem = convertMedicationRequestToLineItem(firstFhirMedicationRequest)
+    convertPrescriptionEndorsements(firstFhirMedicationRequest, hl7v3LineItem)
+    const hl7v3PrescriptionEndorsements = hl7v3LineItem.pertinentInformation3
+
+    expect(hl7v3PrescriptionEndorsements.length).toBeGreaterThan(0)
+
+    hl7v3PrescriptionEndorsements
+      .map(pi3 => expect(pi3.pertinentPrescriberEndorsement.value._attributes.code).toEqual("SLS"))
+  })
+
+  test("are optional for translation", () => {
+    const medicationRequests = getMedicationRequests(bundle)
+
+    const prescriptionEndorsementsFn = (medicationRequest: fhir.MedicationRequest): fhir.CodeableConceptExtension =>
+      getExtensionForUrlOrNull(
+        medicationRequest.extension,
+        "https://fhir.nhs.uk/R4/StructureDefinition/Extension-PrescriptionEndorsement"
+      ) as fhir.CodeableConceptExtension
+
+    medicationRequests.forEach(medicationRequest => {
+      const prescriptionEndorsements = prescriptionEndorsementsFn(medicationRequest)
+      medicationRequest.extension.remove(prescriptionEndorsements)
+      expect(prescriptionEndorsementsFn(medicationRequest)).toEqual(undefined)
+    })
+
+    const hl7v3Prescription = convertBundleToPrescription(bundle)
```

Comment:
i guess its the same again here - can we test this in a way that doesnt convert the entire bundle? unfortunately it seems like the way to do that is to essentially duplicate the code thats written in a function a level above from what we're testing. maybe that means that this test is trying to do more than it should? not sure ive explained what i mean here too well so can jump on call to explain better if needed.

Reply:
yep, lets do that